src/KeyringController.ts

```diff
@@ -442,68 +544,141 @@ class KeyringController extends EventEmitter {
    *
    * Attempts to decrypt the provided message parameters.
    *
-   * @param {object} msgParams - The decryption message parameters.
-   * @param {object} opts - Additional decryption options.
-   * @returns {Promise<Buffer>} The raw decryption result.
+   * @param msgParams - The decryption message parameters.
+   * @returns The raw decryption result.
    */
-  async decryptMessage(msgParams, opts = {}) {
-    const address = normalizeAddress(msgParams.from);
+  async decryptMessage(msgParams: MessageParams): Promise<Bytes> {
+    const address = normalizeAddress(msgParams.from) as Hex;
     const keyring = await this.getKeyringForAccount(address);
-    return keyring.decryptMessage(address, msgParams.data, opts);
+    if (!keyring.decryptMessage) {
+      throw new Error(
+        `KeyringController - The keyring for address ${address} does not support the method decryptMessage.`,
+      );
+    }
+
+    return keyring.decryptMessage(
+      address,
+      msgParams.data as Eip1024EncryptedData,
```

Comment:
This is the error I'm getting,

```
Argument of type 'string | Eip1024EncryptedData | Record<string, unknown>[]' is not assignable to parameter of type 'Eip1024EncryptedData'.

Type 'string' is not assignable to type 'Eip1024EncryptedData'.
```

A conditional should help to avoid the casting.

Reply:
What do you about creating a method to check if the data type is `Eip1024EncryptedData` in `@metamask/utils`? Something like,

```ts
isEip1024EncryptedData(data: unknown): boolean
```

Reply:
If we think it will be applicable to other libraries then it would make sense to put in `utils`, but it seems fairly specific, so maybe placing it in this library would be a good first spot.

In terms of how such a function should be typed, I would probably recommend this instead:

``` typescript
isEip1024EncryptedData(data: unknown): data is Eip1024EncryptedData
```